hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -688,7 +688,10 @@ private void deleteInvalidFilesByPartitions(HoodieEngineContext context, Map<Str
           LOG.info("Deleting invalid data file=" + partitionFilePair);
           // Delete
           try {
-            fileSystem.delete(new Path(partitionFilePair.getValue()), false);
+            boolean result = fileSystem.delete(new Path(partitionFilePair.getValue()), false);
+            if (!result) {
```

Comment:
oh, on re-attempt, do we know if we might try to delete a non-existant file ?

Reply:
maybe it is more safe to check the existence before deletion.

Reply:
Doing fs.exists check everytime may be ineffcient. Why not ignore i.e catch and just log a warning? If the intention is to delete and file does not already exist then what's point of retrying?

Reply:
This may not work as again it could fail due to same set of invalid files (no change in markers).